ARQ-2071 Use more safe serverifo command to check if container runs

vjuranek opened the pull request:
#### Short description of what this resolves:
Use more safe `serverifo` command to check if container runs

#### Changes proposed in this pull request:

- Switches command for checking if container runs from `list` to `serverinfo` as not all session managers implements `list` command (typically for security reasons)


**Fixes**: https://issues.jboss.org/browse/ARQ-2071

bartoszmajsak commented on tomcat-common/src/main/java/org/jboss/arquillian/container/tomcat/TomcatManager.java:
```diff
@@ -100,6 +100,10 @@ public void undeploy(final String name) throws IOException, DeploymentException
+        execute(tomcatManagerCommandSpec.getServerInfoCommand(), null, null, -1);
+    }
+    
     public void list() throws IOException {
```
It seems that by introducinb `serverInfo` `list` method became obsolete. Can you clean up unused code?

MatousJobanek replied:
I think it's same also for the methods `getListCommand()` in `Tomcat**ManagerCommandSpec`  classes

vjuranek commented:
As discussed with @MatousJobanek, to keep backward compatibility, `list()` method should be deprecated first and removed in release after next release. I added deprecated annotation and comment. Is this ok?

bartoszmajsak commented:
Is it something which is public facing? I thought it's more an internal implementation of how we handle the containers (in this case Tomcat). If that is the case, then obviously `@Deprecated` makes more sense.

vjuranek commented:
The method is `public`, so yes, it's public. I can imagine what someone who has some custom extension can rely on this method, but as I'm not very familiar with Arquillian project and its community, project policy etc., I of course can remove it if you like (I just wanted to pop this up for consideration as this is what many projects do - mark public method deprecated in the release after the change and remove it the next release).

bartoszmajsak commented:
I totally understand your concern and appreciate raising it. I'm not aware of any usage of it. I thought maybe @MatousJobanek knows something :)

MatousJobanek commented:
Neither do I TBH :-D. I thought that @vjuranek uses this code somewhere in his test suite (as an API method).
It's really funny this conversation - too many assumptions. We should follow @bartoszmajsak and not to have any expectation and don't assume anything :-D

On the other hand, we have a CR release and according to "standard process" I think that we shouldn't remove (radically change) anything that is publicly available (public methods mainly in interfaces).

Personally, I would keep it with the annotation `@Deprecated` for this time, release finally the Final version and then remove it completely from the code.

bartoszmajsak commented:
+1, squashed and merged! Thank you @vjuranek

vjuranek commented:
great, thanks!
